torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -237,14 +247,24 @@ def __init__(
         self.alpha = alpha
         self.has_bias = has_bias
         self.register_blocking = register_blocking
-        m, n = layout.size
-        _, k = input_nodes[0].get_size()
+        m, n = layout.size[-2:]
+        k = input_nodes[0].get_size()[-1]
         self.m, self.n, self.k = m, n, k
         self.padded_n = get_padded_n(n, self.register_blocking.block_n)
         self.is_dynamic_M = has_free_symbols((m,))
+        self.should_pack_weights = True
```

Comment:
Does it make more sense to generalize this class to support normal unpacked weights - renaming `CppPackedGemmTemplate` to `CppGemmTemplate` that allows both packed and unpacked weights?